definitions/npm/react-redux_v4.x.x/flow_v0.30.x-/react-redux_v4.x.x.js

```diff
@@ -53,7 +53,7 @@ type ConnectDispatchStateless = <P, SP, Dispatch: Function>(
   mapDispatchToProps: null | void,
   mergeProps: null | void,
   options?: {pure?: boolean, withRef?: boolean}
-) => (component: (props: P) => any) => Class<React$Component<void, $Diff<$Diff<P, {dispatch: Dispatch}>, SP>, void>>;
+) => (component: (props: P) => any) => Class<React$Component<void, $Diff<P, SP&{dispatch: Dispatch}>, void>>;
```

Comment:
Yep, I tested it with 0.30, 0.31, and 0.32 and it seems to work as you described.